src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeBodyPartRotationalData.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.immutable.entity;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.flowpowered.math.vector.Vector3d;
+import com.google.common.collect.ImmutableMap;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableBodyPartRotationalData;
+import org.spongepowered.api.data.manipulator.mutable.entity.BodyPartRotationalData;
+import org.spongepowered.api.data.type.BodyPart;
+import org.spongepowered.api.data.type.BodyParts;
+import org.spongepowered.api.data.value.immutable.ImmutableMapValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.common.data.manipulator.immutable.common.AbstractImmutableData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeBodyPartRotationalData;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeMapValue;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+
+import java.util.Map;
+
+public class ImmutableSpongeBodyPartRotationalData extends AbstractImmutableData<ImmutableBodyPartRotationalData, BodyPartRotationalData> 
+    implements ImmutableBodyPartRotationalData {
+
+    private final Map<BodyPart, Vector3d> rotations;
+    private final ImmutableMapValue<BodyPart, Vector3d> rotationsValue;
+    private final ImmutableValue<Vector3d> headRotationValue;
+    private final ImmutableValue<Vector3d> chestRotationValue;
+    private final ImmutableValue<Vector3d> leftArmRotationValue;
+    private final ImmutableValue<Vector3d> rightArmRotationValue;
+    private final ImmutableValue<Vector3d> leftLegRotationValue;
+    private final ImmutableValue<Vector3d> rightLegRotationValue;
+
+    public ImmutableSpongeBodyPartRotationalData(Map<BodyPart, Vector3d> rotations) {
+        super(ImmutableBodyPartRotationalData.class);
+        
+        this.rotations = ImmutableMap.copyOf(checkNotNull(rotations));
+        this.rotationsValue = new ImmutableSpongeMapValue<>(Keys.BODY_ROTATIONS, this.rotations);
+        this.headRotationValue = new ImmutableSpongeValue<>(Keys.HEAD_ROTATION, this.rotations.get(BodyParts.HEAD));
```

Comment:
Would be a good idea to check the BodyPart exists in the map first, fall back to default if not